draft-steele-cose-hash-envelope.md

```diff
@@ -38,88 +38,117 @@ author:
 
 --- middle
 
 # Introduction
 
 COSE defined detached payloads in Section 2 of {{-RFC9052}}, using `nil` as the payload.
-
-In order to verify a signature over a detached payload, the verifier must have access to the payload content. Storing a hash of the content allows for small signature envelopes, that are easy to transport and verify independently.
+In order to verify a signature over a detached payload, the verifier must have access to the payload content.
+Storing a hash of the content allows for small signature envelopes, that are easy to transport and verify independently.
 
 Additional hints in the protected header ensure cryptographic agility for the hashing & signing algorithms, and discoverability for the original content which could be prohibitively large to move over a network.
 
-## Requirements Notation
+## Attached Payload
 
-{::boilerplate bcp14-tagged}
+COSE_sign1 envelope with an attached payload, providing for signature validation.
+
+~~~~ cbor-diag
+18(                                 / COSE Sign 1                   /
+    [
+      h'a4013822...3a616263',       / Protected                     /
+      {}                            / Unprotected                   /
+      h'317cedc7...c494e772',       / Payload                       /
+      h'15280897...93ef39e5'        / Signature                     /
+    ]
+)
+~~~~
+
+## Detached Payload
+
+COSE_sign1 envelope with a detached payload (`nil`), which is compact but the payload must be distributed out of band to validate the signature
+
+~~~~ cbor-diag
+18(                                 / COSE Sign 1                   /
+    [
+      h'a4013822...3a616263',       / Protected                     /
+      {}                            / Unprotected                   /
+      nil,                          / Detached Payload              /
+      h'15280897...93ef39e5'        / Signature                     /
+    ]
+)
+~~~~
+
+## Hashed Payload
+
+A hashed payload functions equivalently to an attached payload, with the benefits of being compact in size and providing the ability to validate the signature.
+
+~~~~ cbor-diag
+18(                                 / COSE Sign 1                   /
+    [
+      h'a4013822...3a616263',       / Protected                     /
+      {}                            / Unprotected                   /
+      h'935b5a91...e18a588a',       / Payload                       /
+      h'15280897...93ef39e5'        / Signature                     /
+    ]
+)
+~~~~
 
 # Header Parameters
 
-To represent a hash of a detached payload, the following headers are defined:
+To represent a hash of a payload, the following headers are defined:
 
-TBD 0:
-  : will be assigned by {{-COSE-TYP}}, represents the content type of the code envelope, including the protected header and payload
+typ 16:
+  : Assigned by {{-COSE-TYP}}, represents the content type of the code envelope, including the protected header and payload
```

Comment:
```suggestion
```

Reply:
We don't even need to define this anymore

Reply:
The section says "Header Parameters", and these are the header parameters. One just happens to already have an assignment. As opposed to, "Requested Header Parameters".

Once these are assigned, they'd all have allocations.

Reply:
I think this suggestion should be applied because:

https://github.com/OR13/draft-steele-cose-hash-envelope/pull/11/files#diff-939f0d001ab3ed3d6adaedfa8e4b077068e26edc5477607a69c04eb6b27e51a5R130

typ is optional in the CDDL and its defined in a different document.